lisa/schema.py

```diff
@@ -678,8 +678,9 @@ class Platform(TypedSchema, ExtendableSchemaMixin):
     admin_password: str = ""
     admin_private_key_file: str = ""
 
-    # True means not to delete an environment, even it's created by lisa
-    reserve_environment: bool = False
+    # No/False: means to delete the environment regardless case fail or pass
+    # Always/True: means to keep the environment regardless case fail or pass
+    reserve_environment: Optional[Union[str, bool]] = False
```

Comment:
For allowed strings, check `validate.OneOf` pattern in the schema. But I'm not sure if it works with Union. If not, try to check in `__post_init__`